### PR TITLE
simple64: add desktop file

### PR DESCRIPTION
**Pull request by Gliczy**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/si/simple64/package.nix [hunk 1]
```diff
@@ -91,6 +94,26 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "simple64";
+      desktopName = "simple64";
+      genericName = "Nintendo 64 Emulator";
+      type = "Application";
```
Comment (TomaSajt) [inline pkgs/by-name/si/simple64/package.nix:102]: This is the default, no need to specify
```suggestion
```

File: pkgs/by-name/si/simple64/package.nix [hunk 2]
```diff
@@ -91,6 +94,26 @@ stdenv.mkDerivation (finalAttrs: {
...
+      exec = "simple64-gui";
+      mimeTypes = [ "application/x-n64-rom" ];
+      icon = fetchurl {
+        url = "https://raw.githubusercontent.com/simple64/simple64/v${finalAttrs.version}/simple64-gui/icons/simple64.svg";
+        hash = "sha256-d5TEq4spbAOQoUKH2COxDUO3DweGBiCQKAS/6izJ54s=";
+      };
```
Comment (TomaSajt) [inline pkgs/by-name/si/simple64/package.nix:108]:
```suggestion
      icon = "simple64";
```
it is not recommended to have store paths in `.desktop` files.

The `.desktop` file automatically search the icon at `$out/share/icons/hicolor/scalable/apps/$icon.svg`, so you should install the `.svg` file to that location in `intallPhase`.

**Comment by Gliczy** [comment]
I wasn't sure how to deal with the icon field, your solution is indeed better.
Thanks for your insight!

File: pkgs/by-name/si/simple64/package.nix [hunk 3]
```diff
@@ -84,13 +87,32 @@ stdenv.mkDerivation (finalAttrs: {
     mkdir -p $out/share/simple64 $out/bin
     cp -r simple64/* $out/share/simple64
 
+    mkdir -p $out/share/icons/hicolor/scalable/apps
+    install -Dm644 ./simple64-gui/icons/simple64.svg $out/share/icons/hicolor/scalable/apps/
```
Comment (TomaSajt) [inline pkgs/by-name/si/simple64/package.nix:91]: nit: there is a way to do this in one line:

```suggestion
    install -Dm644 ./simple64-gui/icons/simple64.svg -t $out/share/icons/hicolor/scalable/apps/
```
or
```suggestion
    install -Dm644 ./simple64-gui/icons/simple64.svg $out/share/icons/hicolor/scalable/apps/simple64.svg
```

(the latter is mostly useful if the file copied needs to have a different name after copying)

**Review by TomaSajt — APPROVED** [review]
LGTM

**Pull request merged**